src/auth/oauth/oauth.ts

```diff
@@ -188,12 +193,19 @@ const ShopifyOAuth = {
       currentSession.scope = responseBody.scope;
     }
 
-    cookies.set(ShopifyOAuth.SESSION_COOKIE_NAME, currentSession.id, {
-      signed: true,
-      expires: Context.IS_EMBEDDED_APP ? new Date() : currentSession.expires,
-      sameSite: 'lax',
-      secure: true,
-    });
+    if (isOnline) {
+      const cookies = new Cookies(request, response, {
+        keys: [Context.API_SECRET_KEY],
+        secure: true,
+      });
+
+      cookies.set(ShopifyOAuth.SESSION_COOKIE_NAME, currentSession.id, {
+        signed: true,
+        expires: Context.IS_EMBEDDED_APP ? new Date() : currentSession.expires,
+        sameSite: 'lax',
+        secure: true,
+      });
+    }
```

Comment:
Nit: Since this now only needs to happen for online sessions, can this be moved up to the bottom of the previous `currentSession.isOnline` check that starts on 160 and ends on 189? Could we put this between 188 and 189?

Reply:
Yep, totally makes sense